pceplib/pcep_msg_objects_encoding.c

```diff
@@ -0,0 +1,1483 @@
+    object_encoders[PCEP_OBJ_CLASS_INTER_LAYER] = pcep_encode_obj_inter_layer;
+    object_encoders[PCEP_OBJ_CLASS_SWITCH_LAYER]= pcep_encode_obj_switch_layer;
+    object_encoders[PCEP_OBJ_CLASS_REQ_ADAP_CAP]= pcep_encode_obj_req_adap_cap;
+    object_encoders[PCEP_OBJ_CLASS_SERVER_IND]  = pcep_encode_obj_server_ind;
+    object_encoders[PCEP_OBJ_CLASS_VENDOR_INFO] = pcep_encode_obj_vendor_info;
+    object_encoders[PCEP_OBJ_CLASS_OF]          = pcep_encode_obj_objective_function;
+
+    /* Decoders */
+    memset(object_decoders, 0, sizeof(object_decoder_funcptr) * MAX_OBJECT_ENCODER_INDEX);
+    object_decoders[PCEP_OBJ_CLASS_OPEN]       =  pcep_decode_obj_open;
+    object_decoders[PCEP_OBJ_CLASS_RP]         =  pcep_decode_obj_rp;
+    object_decoders[PCEP_OBJ_CLASS_NOPATH]     =  pcep_decode_obj_nopath;
+    object_decoders[PCEP_OBJ_CLASS_ENDPOINTS]  =  pcep_decode_obj_endpoints;
+    object_decoders[PCEP_OBJ_CLASS_BANDWIDTH]  =  pcep_decode_obj_bandwidth;
+    object_decoders[PCEP_OBJ_CLASS_METRIC]     =  pcep_decode_obj_metric;
+    object_decoders[PCEP_OBJ_CLASS_ERO]        =  pcep_decode_obj_ro;
+    object_decoders[PCEP_OBJ_CLASS_RRO]        =  pcep_decode_obj_ro;
+    object_decoders[PCEP_OBJ_CLASS_LSPA]       =  pcep_decode_obj_lspa;
+    object_decoders[PCEP_OBJ_CLASS_IRO]        =  pcep_decode_obj_ro;
+    object_decoders[PCEP_OBJ_CLASS_SVEC]       =  pcep_decode_obj_svec;
+    object_decoders[PCEP_OBJ_CLASS_NOTF]       =  pcep_decode_obj_notify;
+    object_decoders[PCEP_OBJ_CLASS_ERROR]      =  pcep_decode_obj_error;
+    object_decoders[PCEP_OBJ_CLASS_CLOSE]      =  pcep_decode_obj_close;
+    object_decoders[PCEP_OBJ_CLASS_LSP]        =  pcep_decode_obj_lsp;
+    object_decoders[PCEP_OBJ_CLASS_SRP]        =  pcep_decode_obj_srp;
+    object_decoders[PCEP_OBJ_CLASS_ASSOCIATION] = pcep_decode_obj_association;
+    object_decoders[PCEP_OBJ_CLASS_INTER_LAYER] = pcep_decode_obj_inter_layer;
+    object_decoders[PCEP_OBJ_CLASS_SWITCH_LAYER]= pcep_decode_obj_switch_layer;
+    object_decoders[PCEP_OBJ_CLASS_REQ_ADAP_CAP]= pcep_decode_obj_req_adap_cap;
+    object_decoders[PCEP_OBJ_CLASS_SERVER_IND]  = pcep_decode_obj_server_ind;
+    object_decoders[PCEP_OBJ_CLASS_VENDOR_INFO] = pcep_decode_obj_vendor_info;
+    object_decoders[PCEP_OBJ_CLASS_OF]          = pcep_decode_obj_objective_function;
+}
+
+/*
+ * The TLVs can have strange length values, since they do not include padding in
+ * the TLV header length, but that extra padding must be taken into account by
+ * the enclosing object by rounding up to the next 4 byte boundary.
+ * Example returned lengths:
+ *   normalize_length(4)  =  4, normalize_length(5)  =  8, normalize_length(6)  =  8, normalize_length(7)  =  8,
+ *   normalize_length(8)  =  8  normalize_length(9)  = 12, normalize_length(10) = 12, normalize_length(11) = 12,
+ *   normalize_length(12) = 12, normalize_length(13) = 13...
+ */
+uint16_t normalize_length(uint16_t length)
```

Comment:
This is a very generic name for a symbol exported by a library. Please make it static or prefix it with `pcep` or something.

Reply:
done